src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFakeImpl.java

```diff
@@ -25,10 +28,33 @@ public void createUrlAlias(UrlAlias urlAlias) {
     return aliases.get(alias);
   }
 
+  /**
+   * Delete the UrlAlias if exists from aliases HashMap.
+   *
+   * @param email of the user to whom the UrlAlias belongs
+   * @param alias the UrlAlias which should be removed
+   *
+   * @throws IllegalArgumentException if no such email or alias
+   */
+
   @Override
   public void deleteUrlAlias(String email, String alias) {
-    // TODO: We should implement it
-    throw new UnsupportedOperationException();
+    UrlAlias foundUrlAlias = null;
```

Comment:
You use the hash map, so you don't need the for statement to find element by alias.
```suggestion
    UrlAlias foundUrlAlias = aliases.get(alias);
```